### PR TITLE
feat: soul component bindings

### PR DESCRIPTION
**Pull request by drewpledger**
## What/Why?
Binding for `button-link` component.

https://github.com/user-attachments/assets/d81956cd-0ee8-4068-aef8-5c8cac3c405f

Binding for `accordions` component.

https://github.com/user-attachments/assets/d9f51968-9ff5-4161-b7f5-17910d6c0203

Binding for `card` component.

https://github.com/user-attachments/assets/280133aa-c3e9-44a3-800d-226b76dc2952

Binding for `carousel` component.

https://github.com/user-attachments/assets/cd99893b-fe9e-4074-9876-5fb9560a94b7

Binding for `card-carousel` component.

https://github.com/user-attachments/assets/e215bfa0-7ecb-4a55-b7e5-69d713bbc529

Binding for `product-carousel` component.

https://github.com/user-attachments/assets/fd060c08-1435-4aca-9e03-56c80f69b1f4

Binding for `slideshow` component.

https://github.com/user-attachments/assets/e91c9d3b-92cc-4f6c-a62d-eddd5f605eda

Binding for `featured-image` component

https://github.com/user-attachments/assets/9acd8225-4e58-41f0-99a6-a5bf103843be

Binding for `products-list` component

https://github.com/user-attachments/assets/46aef3e9-958d-48ed-bb2f-d367c852c4be

Binding for `inline-email-form` component.

CleanShot 2024-11-06 at 17 21 10@2x

### CODE REVIEW

**Comment by jamesqquick** [comment]
@drewpledger can you upload that last video for the card component? It looks like it might not have finished uploading before you submitted?

File: core/vibes/soul/primitives/button-link/button-link.makeswift.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  function MSButtonLink({ link, text, ...props }: MSButtonLinkProps) {
+    return (
+      <ButtonLink href={link.href ?? ''} target={link.target ?? ''} {...props}>
+        {text}
+      </ButtonLink>
+    );
+  },
+  {
+    type: 'primitive-button',
+    label: 'Primitives / Button',
```
Comment (jamesqquick) [inline core/vibes/soul/primitives/button-link/button-link.makeswift.tsx:22]: Should we call this `Primitives / Button Link` to be more specific?

File: core/vibes/soul/primitives/card-carousel/card-carousel.makeswift.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+interface Card {
+  title?: string;
+  imageSrc?: string;
+  imageAlt?: string;
+  link?: { href?: string; target?: string };
+}
+
+interface MSCardCarouselProps {
+  className: string;
+  textContrast: 'light' | 'dark';
```
Comment (jamesqquick) [inline core/vibes/soul/primitives/card-carousel/card-carousel.makeswift.tsx:16]: I'm not seeing an difference when toggling the text contrast from light to dark. Should there be?

File: core/vibes/soul/primitives/card-carousel/card-carousel.makeswift.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    props: {
+      className: Style(),
+      textContrast: Select({
+        options: [
+          { value: 'light', label: 'Light' },
+          { value: 'dark', label: 'Dark' },
+        ],
+        defaultValue: 'dark',
+      }),
+      cards: List({
```
Comment (jamesqquick) [inline core/vibes/soul/primitives/card-carousel/card-carousel.makeswift.tsx:50]: `List` as the name feels a bit generic. Thoughts on using `Slides`?

File: core/vibes/soul/sections/slideshow/slideshow.makeswift.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            title: TextInput({ label: 'Title', defaultValue: 'Slide Title' }),
+            description: TextInput({ label: 'Description', defaultValue: 'Slide Description' }),
+            imageSrc: Image(),
+            imageAlt: TextInput({ label: 'Image Alt', defaultValue: 'Slide Image' }),
+            link: Link({ label: 'Link' }),
+            label: TextInput({ label: 'CTA Label', defaultValue: 'Learn More' }),
+          },
+        }),
+        getItemLabel(slide) {
+          return slide?.title || 'Slide';
```
Comment (jamesqquick) [inline core/vibes/soul/sections/slideshow/slideshow.makeswift.tsx:57]: In the builder, the label is defaulting to `Slide` even though title has the default value of `Slide Title`. Not sure why that default value isn't being picked up initially. 

https://github.com/user-attachments/assets/3803e126-50c3-4831-bbbd-5bf6f7d4cdbd

File: core/vibes/soul/primitives/button-link/button-link.makeswift.tsx (hunk 1 above)
Comment (drewpledger) [reply core/vibes/soul/primitives/button-link/button-link.makeswift.tsx:22]: We've always registered a "link" version of the button, which is the only button registered in apps. Just confirmed with Andrew, we'll just call it Button for this

File: core/vibes/soul/primitives/card-carousel/card-carousel.makeswift.tsx (hunk 2 above)
Comment (drewpledger) [reply core/vibes/soul/primitives/card-carousel/card-carousel.makeswift.tsx:16]: Fixed

File: core/vibes/soul/primitives/card-carousel/card-carousel.makeswift.tsx (hunk 3 above)
Comment (drewpledger) [reply core/vibes/soul/primitives/card-carousel/card-carousel.makeswift.tsx:50]: Fixed

File: core/vibes/soul/sections/slideshow/slideshow.makeswift.tsx (hunk 4 above)
Comment (drewpledger) [reply core/vibes/soul/sections/slideshow/slideshow.makeswift.tsx:57]: Fixed

**Review by jamesqquick — DISMISSED** [review]

**Review by bookernath — DISMISSED** [review]

**Review by bookernath — APPROVED** [review]

**Comment by bookernath** [comment]
image

**Pull request merged**